### DISCUSSION TITLE
metrics in a worker

### DISCUSSION THREAD

**Discussion by RichardWright** [jasnell/piscina#508] [Q&A]
Hi

Is there an established pattern for communicating back from a worker to something like prometheus? or should this just be done via a returned object and handled in the main parent process?

**Comment by metcoder95** [comment]
Hey 👋 
It depends on your needs, overall Piscina provides some insights through histograms that can be pushed to something like `prometheus`. You can communicate with the workers through the parent port to grasp custom metrics per-individual worker, but often it is enough to have these insights from the parent thread.
